mqtt/mqtt-broker/src/persist.rs

```diff
@@ -272,22 +294,68 @@ impl From<ConsolidatedState> for BrokerSnapshot {
     }
 }
 
+/// Actual representation of session state data that is saved to disk.
 #[derive(Deserialize, Serialize)]
 struct ConsolidatedSession {
     client_info: ClientInfo,
     subscriptions: HashMap<String, Subscription>,
-    waiting_to_be_sent: Vec<SimplifiedPublication>,
+    waiting_to_be_sent: VecDeque<PublicationRef>,
+    waiting_to_be_acked: VecDeque<InFlightPublicationRef>,
     last_active: DateTime<Utc>,
 }
 
+/// Represents a queued publication
+/// that has a reference to a payload in `ConsolidatedState`
+/// but not yet in-flight.
+#[derive(Deserialize, Serialize)]
+struct PublicationRef {
```

Comment:
Not sure it is a good name for queued publication. Usually `Ref` suffix is used when there is some owned type `Foo` and `FooRef` which is a reference for `Foo`. For instance [X509](https://docs.rs/openssl/0.10.32/openssl/x509/struct.X509.html) and [X509Ref](https://docs.rs/openssl/0.10.32/openssl/x509/struct.X509Ref.html)

In this case, `PublicationRef` does not reference `Publication`

Reply:
@dmolokanov , any better suggestion? Previous name, IMO, was even more confusing. Every time I get back to this code I have to go over and recall what is responsible for what :)

The only other good name I can think of is `PayloadRef`, since this struct has a reference to the payload. If you don't have strong objections - I'd leave it as is.